cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/golang/glog"
+)
+
+func main() {
+	glog.Info("Kubernetes Elasticsearch logging discovery")
+
+	settings, err := clientcmd.LoadFromFile("/etc/token-admin/kubeconfig")
+	if err != nil {
+		glog.Fatalf("Error loading configuration from /etc/token-admin/kubeconfig: %v", err.Error())
+	}
+
+	config, err := clientcmd.NewDefaultClientConfig(*settings, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		glog.Fatalf("Failed to construct config: %v", err)
+	}
+
+	c, err := client.New(config)
+	if err != nil {
+		glog.Fatalf("Failed to make client: %v", err)
+	}
+
+	// Make a few attempts to look for other Elasticsearch logging pods
+	// since it may take a while for them to show up.
+	var pods *api.PodList
+	count := 0
```

Comment:
Can you add a // TODO: Switch to making a service and reading its endpoints. The endpoints controller does the query & join much more efficiently than it's done here. (Details: If N pods use the pattern shown in this file and there are M pods in the system, load on the system is O(N*M), because this list is linear; if all the pods read the endpoints, load on the system is just O(M) because the endpoints controller amortizes its single lists across all services/pods.)